src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -286,33 +286,42 @@ std::shared_ptr<AbstractOperator> LQPTranslator::_translate_join_node(
   const auto input_left_operator = translate_node(node->left_input());
   const auto input_right_operator = translate_node(node->right_input());
 
-  auto join_node = std::dynamic_pointer_cast<JoinNode>(node);
+  auto join_node = std::static_pointer_cast<JoinNode>(node);
 
   if (join_node->join_mode == JoinMode::Cross) {
     PerformanceWarning("CROSS join used");
     return std::make_shared<Product>(input_left_operator, input_right_operator);
   }
 
-  Assert(join_node->join_predicate(), "Need predicate for non Cross Join");
+  Assert(!join_node->join_predicates().empty(), "Need predicate for non Cross Join");
 
-  /**
-   * Assert that the Join Predicate is simple, e.g. of the form <column_a> <predicate> <column_b>.
-   * We do not require <column_a> to be in the left input though.
-   */
-  const auto operator_join_predicate =
-      OperatorJoinPredicate::from_expression(*join_node->join_predicate(), *node->left_input(), *node->right_input());
-  Assert(operator_join_predicate,
-         "Couldn't translate join predicate: "s + join_node->join_predicate()->as_column_name());
+  std::vector<OperatorJoinPredicate> join_predicates;
+
+  for (const auto& predicate_expression : join_node->join_predicates()) {
+    auto join_predicate =
+        OperatorJoinPredicate::from_expression(*predicate_expression, *node->left_input(), *node->right_input());
+    // Assert that the Join Predicates are simple, e.g. of the form <column_a> <predicate> <column_b>.
+    // We do not require <column_a> to be in the left input though.
```

Comment:
This is a bit confusing (even if it was there before). Maybe "<column_a> and <column_b> must be on separate sides, but <column_a> need not be on the left."